### PR TITLE
Rewrite to use FastAPI instead

### PR DESCRIPTION
**Pull request by t-persson**
### Applicable Issues

fixes: eiffel-community/etos#64
fixes: eiffel-community/etos#65

### Description of the Change

Rewrite ETOS API to use FastAPI instead.

### Alternate Designs

There are other frameworks. This one caught my eye as being simple to understand.
It also boasts being very fast in combination with uvicorn.

### Benefits

Faster API, cleaner structure and a way forward to using the ETOS API as the sole external API in ETOS.

### Possible Drawbacks

FastAPI uses asyncio which is not supported everywhere in ETOS. This is mitigated with a sync_to_async function.
This change also uses libraries that conflict with ETOS Library.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@axis.com>

### CODE REVIEW

**Review by andjoe-axis — APPROVED** [review]
+1

**Review by fredjn — CHANGES_REQUESTED** [review]

File: tests/test_routers.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+        Test steps::
+            1. Send a POST request to root without allow_redirects.
+            2. Verify that status code is 308.
+        """
+        self.logger.info("STEP: Send a POST request to root without allow_redirects.")
+        response = self.client.post("/", allow_redirects=False)
+        self.logger.info("STEP: Verify that status code is 308.")
+        assert response.status_code == 308
+
```
[the remark this answers is not in the document]
Comment (t-persson) [reply tests/test_routers.py:83]: Done

File: tests/test_routers.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+        self.logger.info("STEP: Send a HEAD request to root without allow_redirects.")
+        response = self.client.head("/", allow_redirects=False)
+        self.logger.info("STEP: Verify that status code is 308.")
+        assert response.status_code == 308
+
+    def test_head_on_root_with_redirect(self):
+        """Test that HEAD requests on root return 204 when redirected.
+
+        Approval criteria:
+            - HEAD on root shall return 204.
```
[the remark this answers is not in the document]
Comment (t-persson) [reply tests/test_routers.py]: Done

File: src/etos_api/library/graphql.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        :rtype: dict
+        """
+        async with Client(
+            transport=self.transport,
+            fetch_schema_from_transport=True,
+            execute_timeout=self.etos.debug.default_wait_timeout,
+        ) as session:
+            try:
+                response = await session.execute(gql(query))
+                assert response["artifactCreated"]["edges"]
```
[the remark this answers is not in the document]
Comment (t-persson) [reply src/etos_api/library/graphql.py]: Done

**Review by fredjn — APPROVED** [review]

**Review by nordixinfra — COMMENTED** [review]
- Build Status: **SUCCESS**
- Build URL: [Nordix Jenkins](https://jenkins.nordix.org/job/etos-api-build-merge-ubuntu2004-master/1/)
- Command to pull image:
    `docker pull registry.nordix.org/eiffel/etos-api:dev`
    or
    `podman pull registry.nordix.org/eiffel/etos-api:dev`

**Pull request merged**